src/js/apps/patients/patient/dashboard/dashboard_app.js

```diff
@@ -25,21 +25,68 @@ export default App.extend({
   beforeStart({ patient }) {
     const currentWorkspace = Radio.request('workspace', 'current');
     const states = currentWorkspace.getStates();
-    const filter = { states: states.groupByDone().notDone.getFilterIds() };
+
+    this.filter = { states: states.groupByDone().notDone.getFilterIds() };
 
     return [
-      Radio.request('entities', 'fetch:actions:collection:byPatient', { patientId: patient.id, filter }),
-      Radio.request('entities', 'fetch:flows:collection:byPatient', { patientId: patient.id, filter }),
+      Radio.request('entities', 'fetch:actions:collection:byPatient', { patientId: patient.id, filter: this.filter }),
+      Radio.request('entities', 'fetch:flows:collection:byPatient', { patientId: patient.id, filter: this.filter }),
     ];
   },
 
   onStart(options, actions, flows) {
-    this.collection = new Backbone.Collection([...actions.models, ...flows.models]);
+    // remove these when BE api endpoint `filter[state]=` is fixed
+    const doneActions = actions.filter(action => !action.isDone());
+    const doneFlows = flows.filter(flow => !flow.isDone());
```

Comment:
Without this, we'll need to rethink how we determine if a model needs to be fetched or not.

We currently do a `if (this.collection.get(model))`. But without this filtering ☝️, that check won't work in all scenarios.

Reply:
Why? If it is already in the collection you should never need to fetch it because you are maintaining the data with notifications even if it is not visible

Reply:
Ya I agree, don't think it's needed.